File: LEETCODE/LEETCODE-CPP/1394.cpp
// 1394. Find Lucky Integer in an Array
// https://leetcode.com/problems/find-lucky-integer-in-an-array/

// Method-1 using map
class Solution
{
public:
    int findLucky(vector<int> &arr)
    {
        map<int, int> freq;

        for (int i = 0; i < arr.size(); i++)
        {
            freq[arr[i]]++;
        }

        int maxEle = -1;
        for (auto x : freq)
        {
            if (x.first == x.second)
            {
                maxEle = max(maxEle, x.first);
            }
        }

        return maxEle;
    }
};

File: LEETCODE/LEETCODE-JAVA/1394.java
//1394. Find Lucky Integer in an Array
//https://leetcode.com/problems/find-lucky-integer-in-an-array/

//Method-1 using map
class Solution {
    public int findLucky(int[] arr) {
        Map<Integer,Integer> freq= new HashMap<>();

        for(int i=0;i<arr.length;i++){
            freq.put(arr[i],freq.getOrDefault(arr[i],0)+1);
        }

        int maxEle=-1;
        for(int key:freq.keySet()){
            if(freq.get(key)==key){
                maxEle=key;
            }
        }

        return maxEle;
    }
}